scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/msop.yaml

```diff
@@ -0,0 +1,67 @@
+FileHeader:
+  library_Suffix: 'SO'
+  device_type: 'MSOP'
+
+# By: JEFO
+# Email: jacob.overgaard.andersen@gmail.com
+MSOP-16_3x4mm_P0.5mm:
+  size_source: 'http://www.analog.com/media/en/package-pcb-resources/package/pkg_pdf/ltc-legacy-msop/05081669_A_MS16.pdf'
+  body_size_x:
+    nominal: 3
+    tolerance: 0.102
+  body_size_y:
+    nominal: 4
+    tolerance: 0.102
+  overall_size_x:
+    nominal: 4.9
+    tolerance: 0.152
+  lead_width:
+    minimum: 0.17
+    maximum: 0.27
+  lead_len:
+    nominal: 0.53
+    tolerance: 0.152
+  pitch: 0.5
+  num_pins_x: 0
+  num_pins_y: 8
+
+# By: JEFO
+# Email: jacob.overgaard.andersen@gmail.com
+# 
+# Note: Exposed pad is slightly larger to avoid weird numbers
+MSOP-16-1EP_3x4mm_P0.5mm_EP1.65x2.85mm:
+  size_source: 'https://www.analog.com/media/en/package-pcb-resources/package/pkg_pdf/ltc-legacy-msop/05081667_F_MSE16.pdf'
+  body_size_x:
+    nominal: 3
+    tolerance: 0.102
+  body_size_y:
+    nominal: 4
+    tolerance: 0.102
+  overall_size_x:
+    nominal: 4.9
+    tolerance: 0.152
+  lead_width:
+    minimum: 0.17
+    maximum: 0.27
+  lead_len:
+    nominal: 0.53
+    tolerance: 0.152
+
+  EP_size_x:
+    nominal: 1.65
+    tolerance: 0.102
+  EP_size_y:
+    nominal: 2.85
+    tolerance: 0.102
+  EP_num_paste_pads: [2, 2]
+
+  thermal_vias:
+    count: [2, 3]
+    drill: 0.25
+    paste_via_clearance: 0.1
+    EP_num_paste_pads: [2, 2]
+    paste_avoid_via: False
```

Comment:
with this set to false i suggest to set paste coverage a bit higher (0.7 should be enough)

Reply:
Alright, excellent idea.

Any rule of thumb for what solder paste coverage to use in kicad libraries?

Reply:
according to application nodes by different manufacturers it seems like anything between 50 to 80 percent can give good results depending on your exact process, the stencil thickness and solder used.

We typically simply try to get close to the middle of that range (so 65%) and increase it slightly for footprints where paste overlaps vias. (because one will lose some solder paste in such cases.) I typically used 75% in the past. But now reduced it to 70% as 75 seems a bit excessive. (some application notes suggested a range of 50 to 70 percent coverage so 75 might indeed be too much.)

Reply:
I have for similar parts rounded it to 4mm. For soic and similar parts i think the body size is not that important. A better long term solution would be to make the footprint name generation stuff more powerful. (default pyhton format strings don't really offer the options we need.)
That way we could have the real size given in the yaml file while still getting reasonable footprint names.

Reply:
I think that's a great idea. However, for now what should we decide?

I can either use the exact values from the datasheet and get a long name and a footprint that matches exactly, and then add a 3d-package for it, or we can keep it like this and have a reasonably well fitting footprint with a "decent" name.